OpenRA.Game/Network/UnitOrders.cs

```diff
@@ -188,6 +188,10 @@ internal static void ProcessOrder(OrderManager orderManager, World world, int cl
 
 				case "HandshakeRequest":
 					{
+						// Ignore duplicated handshake requests
+						if (orderManager.HandshakeSent)
+							break;
```

Comment:
Showing an error and disconnecting would make the player aware of an ongoing attack.

Reply:
That would turn a difficult to implement theoretical issue into an easy to implement method to force a player to disconnect against their will.

Reply:
There are much easier ways to disconnect a player than abusing this... There are still enough ways left to crash clients.

Reply:
I'm not aware of any easier ways to disconnect players that don't rely on custom maps. Can you reference a specific issue so we can fix that too?

Reply:
I don't intend to do actual crash testing. But if you are just looking for simple ways to disconnect a player, what about just letting the server close the connection normally or injecting a RST packet? That works even if TLS is used btw.

Reply:
I'm not talking about a MITM situation, which I still consider to be exceedingly unlikely, but players using a modified game client to issue the order over a normal connection using a normal server.

Reply:
Then how can a client let the server send a handshake request to another client in your scenario?

Reply:
The client can issue the order the same way they issue chat. They won't get the response, but that doesn't matter if their goal was to make the other players disconnect - which was my comment above.

Reply:
Then the fix is to prevent the a client from doing that. Handshake orders have a different type so that shouldn't be hard?

Reply:
This would require the server to unpack and parse every order from every client, which it isn't set up to do. I simply don't see the value proposition in implementing two new non-trivial code systems (the server order parsing and a way to trigger a new UI prompt) for the sake of a new dialog message will almost certainly never happen, and that that most players wouldn't understand if it did.

Reply:
What we *can* do, however, is enforce on the client-side that certain orders (handshake, lobbyinfo, etc) come from the server rather than a client.